plugins/catalog-backend/src/next/DefaultCatalogProcessingEngine.ts

```diff
@@ -151,6 +153,29 @@ export class DefaultCatalogProcessingEngine implements CatalogProcessingEngine {
 
           track.markProcessorsCompleted(result);
 
+          if (result.ok) {
+            if (stableStringify(state) !== stableStringify(result.state)) {
+              await this.processingDatabase.transaction(async tx => {
+                await this.processingDatabase.updateEntityCache(tx, {
+                  id,
+                  state: {
+                    ttl: CACHE_TTL,
+                    ...result.state,
```

Comment:
Maybe it's a bit unfortunate to mix the actual data with metadata (ttl). This could be `{ ttl: CACHE_TTL, data: result.state }` instead.